test/services/LegalUnitLinkPatchServiceSpec.scala

```diff
@@ -24,113 +23,241 @@ class LegalUnitLinkPatchServiceSpec extends FreeSpec with Matchers with MockFact
+        }
       }
-    }
 
-    "rejects a patch for any path other than '/children/<some-id>'" in new Fixture {
-      val invalidPathPatch = Seq(Operation(Add, s"/child/$VatRef", JsString(toAcronym(ValueAddedTax))))
+      "signalling when a patch cannot be applied as a result of a conflicting change" in new DeleteFixture {
+        (repository.deleteChildLink _).expects(LegalUnitKey, ValueAddedTax, VatUnitId).returning(
+          Future.successful(EditConflicted)
+        )
 
-      whenReady(patchService.applyPatchTo(LegalUnitKey, invalidPathPatch)) { result =>
-        result shouldBe PatchRejected
+        whenReady(patchService.applyPatchTo(LegalUnitKey, DeleteChildPatch)) { result =>
+          result shouldBe PatchConflicted
+        }
       }
-    }
 
-    "rejects a patch containing a value that is not a string and so cannot be a unit type acronym" in new Fixture {
-      val invalidValuePatch = Seq(Operation(Add, s"/children/$VatRef", JsNumber(42)))
+      "signalling when the target Legal Unit cannot be found" in new DeleteFixture {
+        (repository.deleteChildLink _).expects(LegalUnitKey, ValueAddedTax, VatUnitId).returning(
+          Future.successful(EditTargetNotFound)
+        )
 
-      whenReady(patchService.applyPatchTo(LegalUnitKey, invalidValuePatch)) { result =>
-        result shouldBe PatchRejected
+        whenReady(patchService.applyPatchTo(LegalUnitKey, DeleteChildPatch)) { result =>
+          result shouldBe PatchTargetNotFound
+        }
       }
-    }
 
-    "rejects a patch containing a value that is not the acronym for the VAT unit type" in new Fixture {
-      val invalidValuePatch = Seq(Operation(Add, s"/children/$VatRef", JsString(toAcronym(LegalUnit))))
+      "signalling when the application of the patch fails" in new DeleteFixture {
+        (repository.deleteChildLink _).expects(LegalUnitKey, ValueAddedTax, VatUnitId).returning(
+          Future.successful(EditFailed)
+        )
 
-      whenReady(patchService.applyPatchTo(LegalUnitKey, invalidValuePatch)) { result =>
-        result shouldBe PatchRejected
+        whenReady(patchService.applyPatchTo(LegalUnitKey, DeleteChildPatch)) { result =>
+          result shouldBe PatchFailed
+        }
+      }
+
+      "rejecting" - {
+        /*
+         * Permitting an outright 'remove' operation would not allow us to use a 'checkAndDelete' operation,
+         * and would run the risk of overwriting another user's concurrent changes.
+         */
+        "a patch that does not pair a 'test' operation with the 'remove' operation" in new DeleteFixture {
+          whenReady(patchService.applyPatchTo(LegalUnitKey, Seq(RemoveOperation(s"/children/$VatRef")))) { result =>
+            result shouldBe PatchRejected
+          }
+        }
+
+        "a patch that attempts to remove multiple values" in new DeleteFixture {
+          val invalidPatch = Seq(
+            TestOperation(s"/children/$VatRef", JsString(toAcronym(ValueAddedTax))),
```

Comment:
What about the case where there are two (different) `TestOperation`'s and one `RemoveOperation`?

Reply:
Test Case added.